annotation-solr/src/main/java/eu/europeana/annotation/solr/service/impl/SolrAnnotationServiceImpl.java

```diff
@@ -539,87 +538,90 @@ else if(BodyInternalTypes.isSimpleTagBody(anno.getBody().getInternalType()))
-		String solrQuery = "(" + WebAnnotationModelFields.MOTIVATION + ":\"" + MotivationTypes.SUBTITLING.getOaType() + "\"";
-		solrQuery+=" AND " + SolrAnnotationConstants.TARGET_URI + ":\"" + anno.getTarget().getSource() + "\"";
-		solrQuery+=" AND " + SolrAnnotationConstants.BODY_VALUE_PREFIX + anno.getBody().getLanguage() + ":*"; 
-		solrQuery+=")";
-		return solrQuery;
+	private SolrQuery solrUniquenessQueryCaptions(Annotation anno) {
+      SolrQuery query = new SolrQuery();
+      query.setQuery(SolrAnnotationConstants.TARGET_URI + ":\"" + anno.getTarget().getSource() + "\"");
+      query.addFilterQuery(WebAnnotationModelFields.MOTIVATION + ":\"" + MotivationTypes.CAPTIONING.getOaType() + "\"");
+      return query;
 	}
 	
-	private String solrUniquenessQuerySemanticTagging(Annotation anno) {
-		String solrQuery = "(" + WebAnnotationModelFields.MOTIVATION + ":\"" + MotivationTypes.TAGGING.getOaType() + "\"";
-		solrQuery+=" AND " + SolrAnnotationConstants.TARGET_URI + ":\"" + anno.getTarget().getValue() + "\"";
-		
-		List<String> bodyUris = extractUriValues(anno.getBody());
-	    for (String bodyUri : bodyUris) {
-			solrQuery+=" AND " + SolrAnnotationConstants.BODY_URI + ":\"" + bodyUri + "\""; 
-	    }
-
-		solrQuery+=")";
-		return solrQuery;
+	private SolrQuery solrUniquenessQuerySubtitles(Annotation anno) {
+      SolrQuery query = new SolrQuery();
+      query.setQuery(SolrAnnotationConstants.TARGET_URI + ":\"" + anno.getTarget().getSource() + "\"");
+      query.addFilterQuery(WebAnnotationModelFields.MOTIVATION + ":\"" + MotivationTypes.SUBTITLING.getOaType() + "\"");
+      query.addFilterQuery(SolrAnnotationConstants.BODY_VALUE_PREFIX + anno.getBody().getLanguage() + ":*");
+      return query;
 	}
 	
-	private String solrUniquenessQuerySimpleTagging(Annotation anno) {
-		String solrQuery = "(" + WebAnnotationModelFields.MOTIVATION + ":\"" + MotivationTypes.TAGGING.getOaType() + "\"";
-		solrQuery+=" AND " + SolrAnnotationConstants.TARGET_URI + ":\"" + anno.getTarget().getValue() + "\"";
-		solrQuery+=" AND " + SolrAnnotationConstants.BODY_VALUE + ":\"" + anno.getBody().getValue() + "\"";
-		solrQuery+=")";
-		return solrQuery;
+	private SolrQuery solrUniquenessQuerySemanticTagging(Annotation anno) {
+      SolrQuery query = new SolrQuery();
+      query.setQuery(SolrAnnotationConstants.TARGET_URI + ":\"" + anno.getTarget().getValue() + "\"");
+      query.addFilterQuery(WebAnnotationModelFields.MOTIVATION + ":\"" + MotivationTypes.TAGGING.getOaType() + "\"");
+      List<String> bodyUris = extractUriValues(anno.getBody());
+      //using a single filter here (see please: https://solr.apache.org/guide/8_11/common-query-parameters.html#fq-filter-query-parameter, for more information)
+      String bodyUrisFilter = "";
+      for (int i=0; i<bodyUris.size(); i++) {
+        if(i==0) 
+          bodyUrisFilter += "+" + SolrAnnotationConstants.BODY_URI + ":\"" + bodyUris.get(i) + "\"";
```

Comment:
for maintenance purposes, it is better to not concatenate URIs in the query but simply add new filter for each URI

Reply:
Updated.